cygwin/Platform.c

```diff
@@ -0,0 +1,339 @@
+/*
+htop - cygwin/Platform.c
+(C) 2014 Hisham H. Muhammad
+(C) 2015 Michael McConville
+(C) 2015 David C. Hunt
+(C) 2017,2018 Guy M. Broome
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "cygwin/CygwinMachine.h"
+#include "cygwin/Platform.h"
+
+#include <math.h>
+#include <stdio.h>
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif /* WIN32_LEAN_AND_MEAN */
+#include <windows.h>  // for DWORD
+#undef WIN32_LEAN_AND_MEAN
+
+#include "ClockMeter.h"
+#include "CPUMeter.h"
+#include "DateMeter.h"
+#include "DateTimeMeter.h"
+#include "HostnameMeter.h"
+#include "LoadAverageMeter.h"
+#include "Macros.h"
+#include "MemoryMeter.h"
+#include "MemorySwapMeter.h"
+#include "Meter.h"
+#include "Settings.h"
+#include "SwapMeter.h"
+#include "SysArchMeter.h"
+#include "TasksMeter.h"
+#include "UptimeMeter.h"
+#include "XUtils.h"
```

Comment:
```suggestion
#include "cygwin/Platform.h"

#include <math.h>
#include <stdio.h>

#include "ClockMeter.h"
#include "CPUMeter.h"
#include "DateMeter.h"
#include "DateTimeMeter.h"
#include "HostnameMeter.h"
#include "LoadAverageMeter.h"
#include "Macros.h"
#include "MemoryMeter.h"
#include "MemorySwapMeter.h"
#include "Meter.h"
#include "Settings.h"
#include "SwapMeter.h"
#include "SysArchMeter.h"
#include "TasksMeter.h"
#include "UptimeMeter.h"
#include "XUtils.h"

#include "cygwin/CygwinMachine.h"

#ifndef WIN32_LEAN_AND_MEAN
#define WIN32_LEAN_AND_MEAN
#endif /* WIN32_LEAN_AND_MEAN */
#include <windows.h>  // for DWORD
#undef WIN32_LEAN_AND_MEAN
```

Reply:
> [Suggested change]

hum, i think `windows.h` is a system header

Reply:
Defining those `#ifdef`s inline is fine as is. Just a matter of ordering. Also, for things where there's not just the include itself, those are typically pulled down into their own block. It's not fully obvious, that's why this suggested patch.